tldw_Server_API/app/core/DB_Management/chacha/keyword_store.py

```
@@ -0,0 +1,786 @@
+            limit: Max number of keywords.
+            offset: Number to skip.
+
+        Returns:
+            A list of keyword dictionaries.
+        """
+        return self._db._list_generic_items("keywords", "keyword COLLATE NOCASE", limit, offset)
+
+    def count_keywords(self) -> int:
+        """Return count of active (non-deleted) keywords."""
+        keyword_table = self._db._map_table_for_backend("keywords")
+        query = f"SELECT COUNT(*) AS cnt FROM {keyword_table} WHERE deleted = 0"  # nosec B608
```

Ruff (0.15.11)

[error] Possible SQL injection vector through string-based query construction

(S608)